src/EFCore.MySql/Storage/Internal/WrappedMySqlDataReader.cs

```diff
@@ -18,6 +19,8 @@ public class WrappedMySqlDataReader : DbDataReader
 
         private DbDataReader _reader;
         private bool _disposed;
+        private Dictionary<string, Delegate> compiledExpressionCache = new Dictionary<string, Delegate>();
```

Comment:
Probably makes sense to use a `static` [ConcurrentDictionary](https://msdn.microsoft.com/en-us/library/dd287191(v=vs.110).aspx) here and get rid of the lock